docs/meson.build

```diff
@@ -0,0 +1,176 @@
+fs = import('fs')
+
+maybe_fmt_dep = dependency(
+  'fmt',
+  version: fmt_spec,
+  required: false,
+  disabler: true,
+)
+
+# glob **.md
+docs = [
```

Comment:
Similarly, listing all documentation files manually is prone to maintenance issues.
```meson
docs = [
  'README.md',
  'faq.md',
  ...
]
```
Furthermore, the logic to parse Markdown and extract C++ code duplicates the functionality of `tools/extract_example_code_from_docs.py`. Please reuse the existing Python script (e.g., using `custom_target` or `run_command`) instead of reimplementing the extraction logic in Meson.

Reply:
done in cab69150a3ec29d0c321ebde690c5d9daf626df5 407f60914ef5e67abc135e09e8d075057160c216

Reply:
No because meson requires manually listing files to track dependencies correctly. See https://mesonbuild.com/FAQ.html#but-i-really-want-to-use-wildcards for the reasoning.